win32/agent.py

```diff
@@ -282,10 +280,16 @@ def __init__(self, agentConfig, hostname):
 
     def run(self):
         if self.is_enabled:
+            JMXFetch.clean_exit_file()
             self.jmx_daemon.run()
 
-    def stop(self):
-        pass
+    def terminate(self):
+        """
+        Override `terminate` method to properly exit JMXFetch.
+        """
+        JMXFetch.write_exit_file()
+        self.join()
```

Comment:
Do we need that ? Can we just write the exit file and let jmxfetch closes whenever ? otherwise it might take up to 15 seconds to stop.

We just need to make sure to delete the exit file when starting the agent.